### PR TITLE
Fix clang17 build warning on SUSE Linux platform

### LINKED ISSUE #540: SUSE Linux build error
I try to build this project following `build.md`, but this failed as:
```
[20/3905] Building CXX object contrib/v8-cmake/CMakeFiles/v8_base_without_compiler.dir/v8/src/builtins/setup-builtins-internal.cc.o
FAILED: contrib/v8-cmake/CMakeFiles/v8_base_without_compiler.dir/v8/src/builtins/setup-builtins-internal.cc.o 
/usr/bin/ccache /usr/bin/clang++ --target=x86_64-linux-gnu --sysroot=/home/shylock/Workspace/proton/cmake/linux/../../contrib/sysroot/linux-x86_64/x86_64-linux-gnu/libc -DHAS_RESERVED_IDENTIFIER -DSTD_EXCEPTION_HAS_STACK_TRACE=1 -DV8_CONCURRENT_MARKING -DV8_HAVE_TARGET_OS -DV8_TARGET_ARCH_X64 -DV8_TARGET_OS_LINUX -D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS -I/home/shylock/Workspace/proton/contrib/v8-cmake/v8/include -I/home/shylock/Workspace/proton/build/contrib/v8-cmake/generated -I/home/shylock/Workspace/proton/contrib/v8-cmake/v8 -I/home/shylock/Workspace/proton/contrib/v8-cmake/v8/third_party/zlib -I/home/shylock/Workspace/proton/build/contrib/v8-cmake -I/home/shylock/Workspace/proton/base/glibc-compatibility/memcpy -isystem /home/shylock/Workspace/proton/contrib/llvm-project/libcxx/include -isystem /home/shylock/Workspace/proton/contrib/llvm-project/libcxxabi/include -isystem /home/shylock/Workspace/proton/contrib/libunwind/include --gcc-toolchain=/home/shylock/Workspace/proton/cmake/linux/../../contrib/sysroot/linux-x86_64 -fdiagnostics-color=always -Xclang -fuse-ctor-homing -Wno-enum-constexpr-conversion -fsized-deallocation  -gdwarf-aranges -mssse3 -msse4.1 -msse4.2 -mpclmul -mpopcnt -fasynchronous-unwind-tables -ffile-prefix-map=/home/shylock/Workspace/proton=. -falign-functions=32 -mbranches-within-32B-boundaries   -Wall -Wno-unused-command-line-argument  -fdiagnostics-absolute-paths -fstrict-vtable-pointers -w -O2 -g -DNDEBUG -O3 -g -gdwarf-4  -fno-pie -std=gnu++23   -D OS_LINUX -DV8_ENABLE_WEBASSEMBLY -DV8_ADVANCED_BIGINT_ALGORITHMS -DV8_ENABLE_WASM_SIMD256_REVEC -fno-exceptions -Wno-invalid-offsetof -Wno-deprecated-non-prototype -Werror -nostdinc++ -MD -MT contrib/v8-cmake/CMakeFiles/v8_base_without_compiler.dir/v8/src/builtins/setup-builtins-internal.cc.o -MF contrib/v8-cmake/CMakeFiles/v8_base_without_compiler.dir/v8/src/builtins/setup-builtins-internal.cc.o.d -o contrib/v8-cmake/CMakeFiles/v8_base_without_compiler.dir/v8/src/builtins/setup-builtins-internal.cc.o -c /home/shylock/Workspace/proton/contrib/v8-cmake/v8/src/builtins/setup-builtins-internal.cc
[29/3905] Building CXX object contrib/v8-cmake/CMakeFiles/v8_base_without_compiler.dir/v8/src/extensions/externalize-string-extension.cc.o
ninja: build stopped: subcommand failed.
```
 The build command for one object failed, but this don't display the compile error message, and as I known, ninja will display `command`, `stdout` and `stderr` by default when command failed.

### PR DESCRIPTION
**Pull request by Shylock-Hg**
PR checklist:
- Did you run ClangFormat ?
- Did you separate headers to a different section in existing community code base ?
- Did you surround `proton: starts/ends` for new code in existing community code base ?

Please write user-readable short description of the changes:
Fix some compile warning in clang17.
Close #540

### CODE REVIEW

**Comment by jovezhong** [comment]
Thanks @Shylock-Hg for contributing the improvement for building process on SUSE Linux. I updated the title to make it clear. Since it's in Lunar New Year holiday, the code review could be a bit slower than usual.

**Comment by chenziliang** [comment]
Thanks so much for the PR : may you help do a diff with ClickHouse community code ? We like Proton's code base to be aligned with the ClickHouse OSS as much as we can. For example, in https://github.com/ClickHouse/ClickHouse/blob/master/src/Columns/ColumnDecimal.cpp, it used `my_compressed=std::move(compressed)`

**Comment by Shylock-Hg** [comment]
> Thanks so much for the PR : may you help do a diff with ClickHouse community code ? We like Proton's code base to be aligned with the ClickHouse OSS as much as we can. For example, in https://github.com/ClickHouse/ClickHouse/blob/master/src/Columns/ColumnDecimal.cpp, it used `my_compressed=std::move(compressed)`

Is there a script to do diff? And I plan to rename with prefix `my_`, is it accepted?

File: src/Columns/ColumnDecimal.cpp [hunk 1]
```diff
@@ -386,11 +386,11 @@ ColumnPtr ColumnDecimal<T>::compress() const
 
     const size_t compressed_size = compressed->size();
     return ColumnCompressed::create(data_size, compressed_size,
-        [compressed = std::move(compressed), column_size = data_size, scale = this->scale]
+        [c_compressed = std::move(compressed), column_size = data_size, c_scale = this->scale]
```
Comment (yokofly) [inline src/Columns/ColumnDecimal.cpp]: it's hard to diff, I do not think we have a script to support on this.
basically, when encountering a warning, we intends to make the code change same as clickhouse way. like replace 
```suggestion
        [my_compressed = std::move(compressed), column_size = data_size, my_scale = this->scale]
```

https://github.com/ClickHouse/ClickHouse/blob/daf1ca06253ff4370622aa45befc00a2af19e1d6/src/Columns/ColumnDecimal.cpp#L486-L492

**Comment by chenziliang** [comment]
> > Thanks so much for the PR : may you help do a diff with ClickHouse community code ? We like Proton's code base to be aligned with the ClickHouse OSS as much as we can. For example, in https://github.com/ClickHouse/ClickHouse/blob/master/src/Columns/ColumnDecimal.cpp, it used `my_compressed=std::move(compressed)`
> 
> Is there a script to do diff? And I plan to rename with prefix `my_`, is it accepted?

We don't have a script yet and it is a manual process. I did this by using vimdiff and carefully pick what we need. Yes rename the variable names same as CH community does is  what we like to have here.

**Comment by yokofly** [comment]
Edit v1: add more context
Edit v2: add regex

thanks for continue working on this pr. 

The rest part seems only need to replace all the changed prefix, I have a simple but still complex way. git patch will generate a file to log changed code. consider use regex for the git patch file with vim/vscode such editor, and then git apply under a new branch, finally directly force push a new commit. 

Remember take a look about **git** **reflog** if u wanna back to previous status/commit.

**Comment by yokofly** [comment]
this pr is really clear.
I locally build with ubuntu 20.04 and clang17 on this feature branch works well, and the smoke test and unit test passed.
I will check detailed a bit later.

File: src/Access/AccessChangesNotifier.cpp [hunk 2]
```diff
@@ -47,10 +47,10 @@ scope_guard AccessChangesNotifier::subscribeForChanges(AccessEntityType type, co
     list.push_back(handler);
     auto handler_it = std::prev(list.end());
 
-    return [handlers=handlers, type, handler_it]
+    return [my_handlers=handlers, type, handler_it]
```
Comment (yokofly) [inline src/Access/AccessChangesNotifier.cpp]:
```suggestion
    return [my_handlers = handlers, type, handler_it]
```

File: src/Access/AccessChangesNotifier.cpp [hunk 3]
```diff
@@ -63,13 +63,13 @@ scope_guard AccessChangesNotifier::subscribeForChanges(const UUID & id, const On
     list.push_back(handler);
     auto handler_it = std::prev(list.end());
 
-    return [handlers=handlers, it, handler_it]
+    return [my_handlers=handlers, it, handler_it]
```
Comment (yokofly) [inline src/Access/AccessChangesNotifier.cpp]:
```suggestion
    return [my_handlers = handlers, it, handler_it]
```

File: src/Access/EnabledRoles.cpp [hunk 4]
```diff
@@ -26,10 +26,10 @@ scope_guard EnabledRoles::subscribeForChanges(const OnChangeHandler & handler) c
     handlers->list.push_back(handler);
     auto it = std::prev(handlers->list.end());
 
-    return [handlers=handlers, it]
+    return [my_handlers=handlers, it]
```
Comment (yokofly) [inline src/Access/EnabledRoles.cpp]:
```suggestion
    return [my_handlers = handlers, it]
```

File: src/Client/Suggest.cpp [hunk 5]
```diff
@@ -100,14 +100,14 @@ static String getLoadSuggestionQuery(Int32 suggestion_limit, bool basic_suggesti
 template <typename ConnectionType>
 void Suggest::load(ContextPtr context, const ConnectionParameters & connection_parameters, Int32 suggestion_limit)
 {
-    loading_thread = std::thread([context=Context::createCopy(context), connection_parameters, suggestion_limit, this]
+    loading_thread = std::thread([my_context=Context::createCopy(context), connection_parameters, suggestion_limit, this]
```
Comment (yokofly) [inline src/Client/Suggest.cpp]:
```suggestion
    loading_thread = std::thread([my_context = Context::createCopy(context), connection_parameters, suggestion_limit, this]
```

**Review by yokofly — APPROVED** [review]

**Comment by yokofly** [comment]
overall is good👍 will  force merge this PR  tomorrow.

**Pull request merged**